File: Source/Herc.Pwa.Client/Features/Edge/EdgeCurrencyWallet/UpdateEdgeCurrencyWallet/UpdateEdgeCurrencyWalletAction.cs
namespace Herc.Pwa.Client.Features.Edge.EdgeCurrencyWallet
{
  using System.Collections.Generic;
  using Herc.Pwa.Client.Features.Edge.Dtos;
  using MediatR;

  public class UpdateEdgeCurrencyWalletAction : IRequest<EdgeCurrencyWalletsState>
  {
    public Dictionary<string, string> Balances { get; set; }
    public List<string> EnabledTokens { get; set; }
    public string FiatCurrencyCode { get; set; }
    public string Id { get; set; }
    public Dictionary<string, string> Keys { get; set; }
    public string Name { get; set; }
    public List<EdgeTransactionDto> EdgeTransactions { get; set; }
  }
}

File: Source/Herc.Pwa.Js/source/Edge/Actions/UpdateEdgeCurrencyWalletAction.ts
import { EdgeBalances } from "../TypeDefinitions/EdgeBalances";
import { EdgeTransaction } from "../TypeDefinitions/EdgeTransaction";

export interface UpdateEdgeCurrencyWalletAction {
  fiatCurrencyCode: string;
  id: string;
  keys: any;
  balances: EdgeBalances;
  enabledTokens: Array<string>;
  name: string | null;
  edgeTransactions: Array<EdgeTransaction>;
}
